src/components/Modals/TransactionDetailModal/TransactionDetailModal.tsx

```diff
@@ -30,20 +35,28 @@ const TransactionDetailModal: React.FC<ModalProps> = ({
   switch (transaction.type) {
     case TransactionType.DEPOSIT:
       data = transaction.data as Deposit
+      dataComponent = (
+        <Data label={'tx'}>
+          <ExplorerLink network={Network.ETHEREUM} txHash={data.hash} />
+        </Data>
+      )
       break
     case TransactionType.WITHDRAWAL:
       data = transaction.data as Withdrawal
+      dataComponent = <WithdrawalDataComponent data={transaction.data} />
       break
     case TransactionType.PURCHASE:
       data = transaction.data as Purchase
       break
     case TransactionType.TRANSFER:
       data = transaction.data as Transfer
       dataComponent = (
-        <div className="data">
-          <div> {t('transaction_detail_modal.to')} </div>
-          <div> {data.to} </div>
-        </div>
+        <>
+          <Data label={'to'}>{data.to}</Data>
+          <Data label={'tx'}>
```

Comment:
we should add i18n to these labels, and the ones below too

Reply:
That is actually handled inside the Data component

```tsx
const Data = ({ label, children }: Props) => {
  return (
    <div className="Data">
      <div>{t(`transaction_detail_modal.${label}`)}</div>
      <div>{children}</div>
    </div>
  )
}

```